### DISCUSSION TITLE
Redis Support?

### DISCUSSION THREAD

**Discussion by ghost** [quarkiverse/quarkus-bucket4j#124] [General]
Can we use Redis as a centralized bucket4j db, so that all instances of our application access and store data in a single, central location?

**Comment by rmanibus** [comment]
@sezginserin by default, this extension [produce a Caffeine proxy manager](https://github.com/quarkiverse/quarkus-bucket4j/blob/217f50f4d531324c8b99200fb1a5e1486de99176/runtime/src/main/java/io/quarkiverse/bucket4j/runtime/DefaultProxyManagerProducer.java#L28C1-L31C6)

but you can override this behavior by producing your own.

Bucket4J support 3 different [redis implementations](https://bucket4j.com/8.13.0/toc.html#bucket4j-redis)

Add one of them to your project, then produce the proxy manager using CDI:

```java
    @Produces
    @ApplicationScoped
    ProxyManager<String> proxyManager() {
        // create your redis proxy manager here
    }
```

**Comment by rmanibus** [comment]
For reference, I started a discussion [here](https://github.com/quarkusio/quarkus/issues/32672) a while back to create a direct integration between Quarkus Cache and Quarkus-Bucket4j, but it is still lacking some key functionalities.
